### PR TITLE
AirfRANS: Fourier+4L/256d+no-EMA — Lion optimizer long-training test

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

Previous AirfRANS experiments showed AdamW >> Lion in short training (2-6 epochs). However, short-run comparisons may be misleading: Lion uses sign-based updates and may need more epochs to settle into a good basin. With Fourier+4L/256d and the full 180-min budget (~8+ epochs), does Lion still underperform? Testing both lr=3e-4 (which excelled in TandemFoil Fourier+physics, PR #2473 val=82.65) and lr=5e-4 (to match AdamW's best LR). Understanding Lion's behavior across long training is critical for building a cross-dataset recipe that generalizes.

## Instructions

Run two Lion LR variants in parallel on separate GPUs:

```bash
# Run 1: Lion lr=3e-4 + Fourier+4L/256d
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py \
  --dataset airfrans \
  --airfrans_task full \
  --optimizer lion \
  --lr 3e-4 \
  --cosine_t_max 150 \
  --no-use-ema \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --epochs 999 \
  --wandb_group levi-airfrans-lion &

# Run 2: Lion lr=5e-4 + Fourier+4L/256d
cd target/icml2026 && CUDA_VISIBLE_DEVICES=1 python train.py \
  --dataset airfrans \
  --airfrans_task full \
  --optimizer lion \
  --lr 5e-4 \
  --cosine_t_max 150 \
  --no-use-ema \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --epochs 999 \
  --wandb_group levi-airfrans-lion &

wait
```

**CRITICAL:** `--epochs 999` is required — epochs=2 is the hardcoded default bug.

Report:
- val_primary/surface_mse for each run at end of budget
- Epochs completed per run
- W&B run IDs
- Comparison vs AdamW winner (0.2387): does Lion close the gap with enough epochs?
- At what epoch (if any) does Lion surpass AdamW's 8-epoch result?

## Baseline

- **val_primary/surface_mse:** 0.2387 (AdamW lr=5e-4, PR #2478)
- **test_primary/surface_mse:** 0.2079
- **Baseline config:** Fourier+4L/256d+no-EMA, AdamW lr=5e-4, T_max=150, 8 epochs, 180-min budget
- **Reproduce:** `cd target/icml2026 && python train.py --dataset airfrans --airfrans_task full --optimizer adamw --lr 5e-4 --cosine_t_max 150 --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 256 --model-heads 4 --epochs 999`
- **External target:** ~0.0043 (60x gap remaining)
- **TandemFoil Lion reference:** val=82.65 with Lion lr=3e-4 (PR #2473)

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2510 because ADVISOR: Closing — baselines have shifted dramatically since this was assigned (TandemFoil 82.65→75.59, AirfRANS 0.2357→0.0696, DrivAerML 51.35%→33.65%). Reassigning with a current-generation experiment..

**Comment by tcapelle** [comment]
STUDENT levi:

## Results (completed before PR closure)

### Run 1: Lion lr=3e-4 (W&B run p0o7t68f)
- **val_primary/surface_mse:** 0.2334 (final, epoch 89) — **best: 0.0609 at epoch 64** (-74.5% vs baseline 0.2387)
- **test_primary/surface_mse:** 0.1981 (final)
- **Epochs:** 89 (182 min)
- **Command:** `cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py --dataset airfrans --airfrans-task full --optimizer lion --lr 3e-4 --cosine-t-max 150 --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 256 --model-heads 4 --epochs 999 --agent levi --wandb-name "levi/airfrans-lion-lr3e4" --wandb-group levi-airfrans-lion`
- **First surpasses AdamW baseline (0.2387):** epoch 13 (val=0.2036)

### Run 2: Lion lr=5e-4 (W&B run 5feph1xh)
- **val_primary/surface_mse:** 0.2453 (final, epoch 89) — **best: 0.2051 at epoch 53** (-14.1% vs baseline 0.2387)
- **test_primary/surface_mse:** 0.2022 (final)
- **Epochs:** 89 (183 min)
- **Command:** `cd target/icml2026 && CUDA_VISIBLE_DEVICES=1 python train.py --dataset airfrans --airfrans-task full --optimizer lion --lr 5e-4 --cosine-t-max 150 --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 256 --model-heads 4 --epochs 999 --agent levi --wandb-name "levi/airfrans-lion-lr5e4" --wandb-group levi-airfrans-lion`
- **First surpasses AdamW baseline (0.2387):** epoch 42 (val=0.2060)

### What happened
Both Lion variants **can** beat AdamW given enough epochs, confirming the hypothesis that short-run comparisons were misleading. However, **extreme instability** is the dominant finding:

- **lr=3e-4** hit a remarkable val=0.0609 at epoch 64 (would be SOTA by a huge margin), but catastrophically spiked to 0.70 at epoch 65 and never recovered to that level. Multiple other spikes throughout training.
- **lr=5e-4** was even more unstable — catastrophic spikes >2.0 at epochs 11, 34, 57, 60, and 88. Loss hit 14.75 at epoch 60.
- Both runs oscillated wildly between excellent and terrible val scores, suggesting Lion's sign-based updates are poorly matched to the loss landscape on this task without additional stabilization.
- The cosine schedule (T_max=150) may exacerbate this — as LR cycles back up after its minimum, it triggers instability in Lion.

### Suggested follow-ups
- **Checkpointing:** The best-epoch results (0.0609, 0.2051) are dramatically better than final results — model checkpointing at best val would capture these
- **Gradient clipping:** May stabilize Lion's sign-based updates
- **Lower T_max:** Try T_max=30 (which worked well for DrivAerML) to keep LR in a more stable range
- **EMA as stabilizer:** Even though no-EMA helps AdamW, Lion's instability might benefit from EMA smoothing
